### PR TITLE
Item 8897: Compliance settings option to hide page background content on invalid session modal

### PR DESCRIPTION
**Pull request by cnathe**
#### Rationale
We are adding in a compliance setting which will hide/blur the background page content when the session invalid modal is shown. This PR adds that updated behavior to the WebSocket.js file along with adding in the compliance moduleContext and ComplianceService updates needed to get this setting info to the client side to conditionalize behavior accordingly.

#### Related Pull Requests
* https://github.com/LabKey/labkey-ui-components/pull/535
* https://github.com/LabKey/compliance/pull/114
* https://github.com/LabKey/platform/pull/2284
* https://github.com/LabKey/biologics/pull/894
* https://github.com/LabKey/sampleManagement/pull/578
* https://github.com/LabKey/inventory/pull/246

#### Changes
* Add "compliance" JSON info to api moduleContext via ComplianceService
* Return require modules (api and internal) for root container call to getActiveModules
* Utils.js updates for modal display (prop to optionally remove close button, support for showModal in LKS and apps
* WebSocket.js updates for compliance setting behavior (automatically hide session invalid modal when user logs back in, add check for isSessionInvalidBackgroundHideEnabled(), change openWebsocket() call to be an opt-in from LK pages, expose initWebSocket on LABKEY.WebSocket)
* core package.json @labkey/components and @labkey/theme package updates
* pageTemplate.jsp update to call initWebSocket for non-app LKS pages

### CODE REVIEW

File: api/webapp/clientapi/dom/WebSocket.js [hunk 1]
```diff
@@ -112,36 +153,51 @@ LABKEY.WebSocket = new function ()
...
     };
 
-    // initial call open the WebSocket to at least handle the logout and session timeout events
-    // other apps or code can register their own event listeners as well
-    openWebsocket();
+    // initial call will open the WebSocket to at least handle the logout and session timeout events
+    // other apps or code can register their own event listeners as well via addServerEventListener
+    var initWebSocket = function() {
+        if ('WebSocket' in window) {
+            if (null === _websocket) {
```
Comment (labkey-alan) [inline api/webapp/clientapi/dom/WebSocket.js]:
```suggestion
        if ('WebSocket' in window && null === _websocket) {
```

File: api/webapp/clientapi/dom/WebSocket.js [hunk 2]
```diff
@@ -102,8 +105,46 @@ LABKEY.WebSocket = new function ()
...
+    function isSessionInvalidBackgroundHideEnabled() {
+        if (LABKEY.moduleContext.api.compliance) {
+            return LABKEY.moduleContext.api.compliance.sessionInvalidBackgroundHideEnabled
+        }
+        return false;
+    }
+
+    function toggleBackgroundVisible(shouldBlur) {
+        if (isSessionInvalidBackgroundHideEnabled()) {
+            var divClsSelectors = ['.lk-header-ct', '.lk-body-ct', '.footer-block', '.x4-window', '.x-window', "div[role='dialog'] > .modal"];
```
Comment (labkey-alan) [inline api/webapp/clientapi/dom/WebSocket.js:119]: Not a huge deal, but we can use let/const in our "old school" JS files.

**Review by labkey-alan — APPROVED** [review]
Pointed out some minor nits, but this looks good to merge.

File: api/webapp/clientapi/dom/WebSocket.js (hunk 2 above)
Comment (labkey-nicka) [reply api/webapp/clientapi/dom/WebSocket.js:119]: Kind of depends actually. I'm not sure our "current" (it's really old) minify tooling supports it even though modern browsers do. Just something you might run into if you attempt to do this.

File: api/webapp/clientapi/dom/WebSocket.js (hunk 1 above)
Comment (cnathe) [reply api/webapp/clientapi/dom/WebSocket.js]: done

**Pull request merged**